src/main/java/gift/kakao/service/KakaoServiceImpl.java

```diff
@@ -0,0 +1,59 @@
+package gift.kakao.service;
+
+import gift.kakao.dto.KakaoTokenResponseDto;
+import gift.kakao.dto.KakaoUserResponseDto;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.MediaType;
+import org.springframework.stereotype.Service;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.client.RestClient;
+
+@Service
+public class KakaoServiceImpl implements KakaoService {
+    private final RestClient restClient;
+
+    @Value("${kakao.client-id}")
+    private String clientId;
+
+    @Value("${kakao.redirect-uri}")
+    private String redirectUri;
+
+    @Value("${kakao.token-uri}")
+    private String tokenUri;
+
+    @Value("${kakao.user-info-uri}")
+    private String userInfoUri;
+
+    public KakaoServiceImpl() {
+        this.restClient = RestClient.create();
```

Comment:
```suggestion
            client = RestClient.builder()
            .baseUrl("https://kauth.kakao.com")
            .build();
```
클라이언트를 선언할 때 공통되는 부분들은 이렇게 미리 설정하는 방식을 많이 활용합니다!
지금은 그럴 필요가 없지만 RestClient 활용법에 대해 미리 알아두시기만 하면 좋을 것 같아요!
https://docs.spring.io/spring-framework/reference/integration/rest-clients.html

Reply:
지적 감사드립니다😊